### PR TITLE
Optimize site creation generated SQL

### PR DESCRIPTION
**Pull request by jmendeza**
Optimize site creation generated SQL
https://github.com/craftersoftware/craftercms/issues/1016
https://github.com/craftercms/craftercms/issues/7326

### CODE REVIEW

File: src/main/java/org/craftercms/studio/api/v2/repository/ContentRepository.java [hunk 1]
```diff
@@ -246,14 +247,15 @@ default String getGlobalRepoLastCommitId() {
     String getLastEditCommitId(String siteId, String path);
 
     /**
-     * Get a list of paths that changed since the commit ID provided to commit ID provided
+     * Execute consumers for all site paths for the given site
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/api/v2/repository/ContentRepository.java:250]: Can we discuss this one?

File: src/main/java/org/craftercms/studio/impl/v1/service/site/SiteServiceImpl.java [hunk 2]
```diff
@@ -138,8 +142,9 @@
 public class SiteServiceImpl implements SiteService, ApplicationContextAware {
 
     private final static Logger logger = LoggerFactory.getLogger(SiteServiceImpl.class);
-    private final static String UPDATE_PARENT_ID_SCRIPT_PREFIX = "updateParentId_";
     private final static String CREATED_FILES_SCRIPT_PREFIX = "createdFiles_";
+    // Max number of items to insert using a single batch
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v1/service/site/SiteServiceImpl.java]: Max items to insert in a single batch

File: src/main/java/org/craftercms/studio/impl/v1/service/site/SiteServiceImpl.java [hunk 3]
```diff
@@ -240,92 +245,83 @@ public void createSiteFromBlueprint(
...
-                Map<String, String> createdFiles =
-                        contentRepositoryV2.getChangeSetPathsFromDelta(siteId, null, lastCommitId);
+            // Add default groups
+            logger.debug("Add the default groups to site '{}'", siteName);
+            addDefaultGroupsForNewSite();
 
-                logger.debug("Add audit log to site '{}'", siteName);
-                insertCreateSiteAuditLog(siteId, siteName, blueprintId, creator);
+            logger.debug("Add audit log to site '{}'", siteName);
+            insertCreateSiteAuditLog(siteId, siteName, blueprintId, creator);
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v1/service/site/SiteServiceImpl.java:280]: What does this do?

File: src/main/java/org/craftercms/studio/impl/v1/service/site/SiteServiceImpl.java [hunk 4]
```diff
@@ -240,92 +245,83 @@ public void createSiteFromBlueprint(
...
 
-                TimeUtils.logExecutionTime(() -> {
-                    processCreatedFiles(siteId, createdFiles, creator, now);
-                    return null;
-                }, logger, format("Method 'SiteServiceImpl.processCreatedFiles(..)' with parameters %s", Arrays.asList(siteId, createdFiles, creator, now)));
+            TimeUtils.logExecutionTimeThrowing(() -> {
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v1/service/site/SiteServiceImpl.java]: Let's discuss this.

File: src/test/java/org/craftercms/studio/impl/v2/repository/blob/BlobAwareContentRepositoryTest.java [hunk 5]
```diff
@@ -115,10 +114,15 @@ public void setUp() throws Exception {
         when(localV1.getContent(SITE, POINTER_PATH)).thenReturn(POINTER);
         when(localV1.isFolder(SITE, PARENT_PATH)).thenReturn(true);
 
-        Map<String, String> delta = new TreeMap<>();
-        delta.put(POINTER_PATH, "D");
-        delta.put(NEW_POINTER_PATH, POINTER_PATH);
-        when(localRepositoryV2.getChangeSetPathsFromDelta(SITE, COMMIT_1, COMMIT_2)).thenReturn(delta);
+//        when(localRepositoryV2.getAllSitePaths(SITE)).thenReturn(
```
Comment (sumerjabri) [inline src/test/java/org/craftercms/studio/impl/v2/repository/blob/BlobAwareContentRepositoryTest.java]: Why keep this commented out stuff?

**Review by sumerjabri — CHANGES_REQUESTED** [review]

File: src/main/java/org/craftercms/studio/impl/v1/service/site/SiteServiceImpl.java (hunk 2 above)
Comment (jmendeza) [reply src/main/java/org/craftercms/studio/impl/v1/service/site/SiteServiceImpl.java]: Updated

File: src/test/java/org/craftercms/studio/impl/v2/repository/blob/BlobAwareContentRepositoryTest.java (hunk 5 above)
Comment (jmendeza) [reply src/test/java/org/craftercms/studio/impl/v2/repository/blob/BlobAwareContentRepositoryTest.java]: Updated

**Review by sumerjabri — APPROVED** [review]

**Pull request merged**